muspinsim_gen/muspinsim_gen.xml

```diff
@@ -0,0 +1,116 @@
+                year = {2023}
+                doi = {10.5281/zenodo.7704635}
+            }
+        </token>
+    </macros>
+    <creator>
+        <person givenName="Joel" familyName="Davies" url="https://github.com/joelvdavies" identifier="https://orcid.org/0000-0002-4153-6819"/>
+        <organization url="https://muon-spectroscopy-computational-project.github.io/index.html" name="The Muon Spectroscopy Computational Project"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">muspinsim</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        structure_name_internal=\$(sed 's/ /\_/g' <<< '$structure_file.name') &&
+        ln -s '$structure_file' \$structure_name_internal &&
+        muspinsim-gen \$structure_name_internal $number_closest --out muspinsim_gen_out.in
+        #if $dipolar:
+            --dipolar
+        #end if
+        #if $quadrupolar_conditional.quadrupolar:
+            --quadrupolar
+            #if $quadrupolar_conditional.gipaw_out_file:
+                $quadrupolar_conditional.gipaw_out_file
+            #end if
+        #end if
+        #if $include_interatomic:
+            --include_interatomic
+        #end if
+        --muon_symbol $muon_symbol
+        #for $ignored_symbol in $ignored_symbols:
+        --ignore_symbol $ignored_symbol.ignored_symbol
+        #end for
+        --max_layer $max_layer
+    ]]></command>
+    <inputs>
+        <param type="data" name="structure_file" format="cif,cell,magres" label="Structure file (.cif, .cell or .magres)"/>
+        <param type="integer" name="number_closest" value="8" label="Number of closest nuclei" help="Number of closest nuclei to the muon which should included in the generated file"/>
+        <param type="boolean" name="dipolar" checked="true" label="Include dipole interactions?"/>
+        <conditional name="quadrupolar_conditional">
+            <param type="boolean" name="quadrupolar" checked="false" label="Include quadrupole interactions?" help="Either requires a magres input file with EFG tensors calculated via CASTEP, or the output from Quantum ESPRESSO's GIPAW program which can be specified once selected."/>
+            <when value="true">
+                <param type="data" name="gipaw_out_file" format="data" optional="true" label="GIPAW Output File (Optional)" help="File containing the output of Quantum ESPRESSO's GIPAW program. Only required if the structure file is not a magres file containing EFG tensors."/>
+            </when>
+            <when value="false">
+            </when>
+        </conditional>
+        <param type="boolean" name="include_interatomic" checked="false" label="Include interatomic interactions?" help="Whether to include interactions between selected nuclei which don't involve the muon."/>
+        <repeat name="ignored_symbols" title="Ignored symbols" min="0" help="Specify symbols that should be ignored e.g. if they have zero spin">
```

Comment:
This is not an issue with the wrapper, but when trying to see what would happen when ignoring all the symbols in the file, the error you get is just
```
  File "/home/kez92691/muon-galaxy/.venv/lib/python3.8/site-packages/muspinsim/input/structure.py", line 323, in compute_closest
    closest_new_atom = new_atoms[0]
IndexError: list index out of range
```
Could we add a check earlier that gives a more helpful error message (in MSS directly, not here). This is my fault for not spotting in the review, I remember thinking about what would happen if we skip all the atoms and expand the cell infitinely (hence why `max_layer` seemed reassuring), but clearly I never actually checked to see what it would do.